warehouse/query-core/src/main/java/datawave/query/function/JexlContextCreator.java

```diff
@@ -64,7 +65,47 @@ public JexlContextCreator() {
             log.trace("Constructed context from index and attribute Documents: " + context);
         }
         
-        return Tuples.tuple(from.first(), from.second(), context);
+        return Tuples.tuple(dedupedFrom.first(), dedupedFrom.second(), context);
+    }
+    
+    private Tuple3 dedupeAttributes(Tuple3<Key,Document,Map<String,Object>> rawTuple) {
+        Map<String,Attribute<? extends Comparable<?>>> dict = rawTuple.second().getDictionary();
+        Set<Attribute<? extends Comparable<?>>> dedupedAttributeSet = new HashSet<Attribute<? extends Comparable<?>>>();
+        
+        for (Entry<String,Attribute<? extends Comparable<?>>> entry : dict.entrySet()) {
+            if (entry.getKey().equals(BODY_KEY)) {
+                Attributes rawButes = (Attributes) entry.getValue();
+                Set<Attribute<? extends Comparable<?>>> attributeSet = rawButes.getAttributes();
+                List<String> valueList = new ArrayList<String>();
+                
+                // get list of any non-preNormalizedAttributes since those are the ones
+                // we actually care about and want to keep
+                attributeSet.forEach(attribute -> {
```

Comment:
I think we should use the Attribute.isFromIndex() instead of checking for PreNormalizedAttribute.  That would be much less fragile.